### ISSUE TITLE
All repos should test against lowest deps, and maybe dev deps

### ISSUE THREAD

**Issue by stof** [minkphp/Mink#667]
We should have a job in the matrix testing against lowest dependencies to ensure that the lowest bound is right.

Depending on what they depend on, it might be worth testing against dev versions too (even though we always test them against the dev version of Mink).
You can see https://github.com/jcalderonzumba/MinkPhantomJSDriver/pull/36 for the way to make such setup (the PR contains other enhancements too)

**Comment by aik099** [comment]
Interesting topic, but doing so we'll be testing the dependencies instead of library itself. If we believe SemVer, then testing against lowest version isn't needed.

**Comment by stof** [comment]
no, we will be testing that our lowest bounds are right. Otherwise, it is very easy to say we are compatible with twig 1.0+ for instance, while we actually depend on a feature added in twig 1.8 (this is an example taken from the PR linked above).

And for the dev stability, this is meant to ensure we are compatible with the next version of our dependencies, to catch issues coming from their update (which should not happen if they follow semver, but this still happens sometimes by mistakes)

**Comment by aik099** [comment]
Ah, so we test against lowest dependency version instead of latest dependency version. In that case yes, I agree, that it's needed.

**Comment by stof** [comment]
@aik099 this is what ``--prefer-lowest`` is about in Composer

**Comment by aik099** [comment]
I though you want to test lowest version together with highest versions :)

**Comment by stof** [comment]
I don't understand what you mean ?

**Comment by stof** [comment]
note that I sent the PR against the Mink repo already. PRs against drivers will come later

**Comment by aik099** [comment]
You said: let's test against lowest version of dependencies.
I understood as: let's test against highest version of dependencies (as we do now) + test against lowest version of dependencies.
What you actually meant: let's test against lowest version of dependencies instead of highest version of dependencies.

**Comment by stof** [comment]
no, we should do both. We should have 1 job using the lowest deps, but we should also have jobs using the latest ones (most of our users will actually use the latest ones, so it is important to use them).

**Comment by aik099** [comment]
It's clear now. I was afraid that testing against lowest dependency version would double travis job count, but since it will only add one job, then it's ok.

**Comment by stof** [comment]
yeah, testing lowest deps against all PHP versions would be overkill. We are not trying to test our deps on multiple versions (they should be doing it), and we are likely to execute the same code paths in our own code. What I generally do is using our older PHP version for the lowest deps build.

**Comment by stof** [comment]
Actually, I changed my mind here. We regularly fix bugs in other libraries spotted thanks to our testsuite (in Goutte for instance), and this would force us to be very strict about the lowest bound to be sure that *all* our tests are passing, even the tests for edge cases. This could make our requirements much more complex.
I'm not sure it is worth it.

**Comment by aik099** [comment]
You mention `Added testing against lowest version of dependencies to ensure we got the lower bounds right` in release notes on https://github.com/minkphp/Mink/releases/tag/v1.7.0 , but at same time say that you no longer want to test lowest dependency versions? Are release notes outdated?

**Comment by stof** [comment]
@aik099 this issue is about driver repos, not about Mink itself

**Comment by aik099** [comment]
OK.
